### PR TITLE
Fix #18479, path completion of paths with ampersand characters.

### LINKED ISSUE #18479: Ampersand in folder name breaks auto-completion in CLI
when there is ampersand in folder name e.g.
`include("/Users/sample/data&results/` 
auto-completion stops at this point

affects `0.4.6, 0.5.0-rc4, 0.6.0-dev.601`

### PR DESCRIPTION
**Pull request by dhoegh**
This supersedes #18567 without the controversial part.

### CODE REVIEW

**Comment by dhoegh** [comment]
Bump

File: test/replcompletions.jl [hunk 1]
```diff
@@ -637,6 +637,26 @@ let #test that it can auto complete with spaces in file/path
...
+        #Test for #18479
+        for c in "'`@\$;&"
+            test_dir = "test$(c)test"
+            mkdir(joinpath(path, test_dir))
+            try
+                if !(c in ['\'','$']) # As these characters hold special meaning
+                    # in shell commands the shell path completion cannot complete
+                    # paths with these characters
+                    c,r,res = test_scomplete(test_dir)
+                    @test c[1] == test_dir*(is_windows() ? "\\\\" : "/")
```
Comment (tkelman) [inline test/replcompletions.jl]: `Sys.iswindows()`

Comment (tkelman) [inline test/replcompletions.jl]: no underscore

File: test/replcompletions.jl [hunk 2]
```diff
@@ -647,11 +647,11 @@ let #test that it can auto complete with spaces in file/path
                     # in shell commands the shell path completion cannot complete
                     # paths with these characters
                     c,r,res = test_scomplete(test_dir)
-                    @test c[1] == test_dir*(is_windows() ? "\\\\" : "/")
+                    @test c[1] == test_dir*(Sys.is_windows() ? "\\\\" : "/")
```
Comment (StefanKarpinski) [inline test/replcompletions.jl]: The new version is also without the underscore.

**Review by StefanKarpinski — CHANGES_REQUESTED** [review]

**Comment by dhoegh** [comment]
I have fixed the `Sys.iswindows` rename.

**Review by StefanKarpinski — APPROVED** [review]

**Comment by StefanKarpinski** [comment]
Failures were the now-ubiquitous timeout/arnoldi test combo.

**Pull request merged**